modules/stream/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/stream/StreamCoordinator.scala

```diff
@@ -2,141 +2,197 @@ package ch.epfl.bluebrain.nexus.sourcing.stream
 import shapeless.Typeable
 
 import scala.concurrent.Future
 
 /**
-  * Actor implementation that builds and manages a stream ([[RunnableGraph]]).
+  * Stream coordinator that wraps the actor builds and manages a stream.
   *
-  * @param init   an initialization function to be run when the actor starts, or when the stream is restarted
-  * @param source an initialization function that produces a stream from an initial start value
+  * @param actor the underlying actor.
   */
-class StreamCoordinator[F[_], A: Typeable](init: F[A], source: A => Source[A, _])(implicit sc: Scheduler, F: Effect[F])
-    extends Actor
-    with ActorLogging {
-
-  private val A                              = implicitly[Typeable[A]]
-  private implicit val as: ActorSystem       = context.system
-  private implicit val mt: ActorMaterializer = ActorMaterializer()
-  private var state: Option[A]               = None
-
-  private def initialize(): Unit = {
-    val logError: PartialFunction[Throwable, F[Unit]] = {
-      case err =>
-        log.error(err, "Failed on initialize function with error '{}'", err.getMessage)
-        F.raiseError(err)
-    }
-    val _ = F.toIO(init.map(Start).onError(logError)).unsafeToFuture() pipeTo self
-  }
-
-  override def preStart(): Unit = {
-    super.preStart()
-    initialize()
-  }
-
-  private def buildStream(a: A): RunnableGraph[(UniqueKillSwitch, Future[Done])] = {
-    source(a)
-      .viaMat(KillSwitches.single)(Keep.right)
-      .map { latest =>
-        state = Some(latest)
-        latest
-      }
-      .toMat(Sink.ignore)(Keep.both)
-  }
+class StreamCoordinator[F[_], A](val actor: ActorRef)(implicit F: Effect[F], config: SourcingConfig) {
 
-  override def receive: Receive = {
-    case Start(any) =>
-      A.cast(any) match {
-        case Some(a) =>
-          log.info(
-            "Received initial start value of type '{}', with value '{}' running the indexing function across the element stream",
-            A.describe,
-            a)
-          state = Some(a)
-          val (killSwitch, doneFuture) = buildStream(a).run()
-          doneFuture pipeTo self
-          context.become(running(killSwitch))
-        // $COVERAGE-OFF$
-        case _ =>
-          log.error("Received unknown initial start value '{}', expecting type '{}', stopping", any, A.describe)
-          context.stop(self)
-        // $COVERAGE-ON$
-      }
-    // $COVERAGE-OFF$
-    case Stop =>
-      log.info("Received stop signal while waiting for a start value, stopping")
-      context.stop(self)
-    // $COVERAGE-ON$
+  private implicit val askTimeout: Timeout = config.askTimeout
 
-    case FetchLatestState => sender() ! LatestState(state)
+  /**
+    * Fetches the latest state from the underlying actor [[StreamCoordinatorActor]] .
+    *
+    * @return latest state wrapped in [[F]]
+    */
+  def state(): F[Option[A]] = IO.fromFuture(IO(actor ? FetchLatestState)).to[F].map {
+    case st: LatestState[A] => st.state
   }
```

Comment:
```scala
.recoverWith {
      case th: TimeoutException =>
        log.error(th, "Timed out while fetching state")
        F.raiseError(...)
      case NonFatal(th) =>
        log.error(th, s"Error while fetching state")
        F.raiseError(...)
    }
```
And maybe we need some `StreamCoordinator` errors

Reply:
won't we handle these error at a higher level in KG and `InternalErrors`?

Reply:
In sourcing it is done this way. You can check in `AkkaAggregate` and `AkkaAggregateActor` in order to be consistent.